stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/ChatDomainEventDomainImplTest.kt

```diff
@@ -3,194 +3,244 @@ package io.getstream.chat.android.livedata
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import com.google.common.truth.Truth
 import io.getstream.chat.android.livedata.utils.getOrAwaitValue
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.runBlocking
-import org.junit.Ignore
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runBlockingTest
 import org.junit.Test
 import org.junit.runner.RunWith
 
 /**
  * Verify that all events correctly update state in room
  */
 @RunWith(AndroidJUnit4::class)
-internal class ChatDomainEventDomainImplTest : BaseConnectedIntegrationTest() {
+internal class ChatDomainEventDomainImplTest : BaseDomainTest2() {
 
     @Test
-    fun defaultConfig() = runBlocking(Dispatchers.IO) {
+    fun `verify that a missing channel config returns the default`() = testIOScope.runBlockingTest {
```

Comment:
We can omit scope declaration in tests and use only `runBlockingTest`

Reply:
If you have a look at `ExampleBaseDomainTest2` you'll notice that I use a testIOScope.launch, i'm not quite sure why that's needed, but if i change it into a simple launch or remove it the test fails with the error: "java.lang.IllegalStateException: This job has not completed yet"

![image](https://user-images.githubusercontent.com/265409/98616312-c9828380-22b9-11eb-96dc-437abd9180f8.png)

Reply:
Indeed. I spent some time but couldn't reveal the cause. Seems some setup of BaseDomainTest2 runs async jobs that aren't completed when test gets executed. We could fix it later